### PR TITLE
Update space check

### PR DESCRIPTION
**Pull request by wenyuzhao**
* Record AllocationSpace instead of MMTk space type
* Use recorded AllocationSpace data for `InSpace` checks
* Reduce code duplication.

_This code (as well as all archive-related code) should be removed in the future. Objects of different allocation type should go to different MMTk Spaces._

---

This CL also updates CI scripts. The v8-binding is now able to run all the v8 tests with NoGC.

### CODE REVIEW

File: v8/third_party/heap/mmtk/mmtk.cc [hunk 1]
```diff
@@ -117,7 +117,15 @@ AllocationResult Heap::Allocate(size_t size_in_bytes, AllocationType type, Alloc
   int space = (type == AllocationType::kCode) ? 3 : (type == AllocationType::kReadOnly) ? 4 : (large_object) ? 2 : 0;
   Address result =
       reinterpret_cast<Address>(alloc(tph_mutator_, size_in_bytes, align_bytes, 0, space));
-  tph_archive_insert(tph_data_->archive(), reinterpret_cast<void*>(result), tph_data_->isolate(), uint8_t(space));
+  AllocationSpace allocation_space;
```
Comment (qinsoon) [inline v8/third_party/heap/mmtk/mmtk.cc:120]: If I understand correct, the `space` above is actually the `AllocationSemantics` in mmtk, and this `allocation_space` is the space from V8 runtime. Is this correct? The reason why you would need to save the `allocation_space` for each address result is that mmtk-core does not provide a function to check whether an address is in a certain space, is this right?

Comment (wenyuzhao) [reply v8/third_party/heap/mmtk/mmtk.cc:120]: Yes, `space` is MMTk `AllocationSemantics` and `allocation_space` is the v8 space that the runtime expects the allocation to happen.

The reason to save `allocation_space` is that v8 wants to check `allocation_space` later after allocation. This usually happens in unit tests -- a lot of them add explicit checks to assert some object is located in code space or old space. Until we add all of these spaces to MMTk common plan, I have to store them in a table for later lookups.

(some compiler optimizations and heap serialization code also want to check the `allocation_space` for a given object.)

Comment (qinsoon) [reply v8/third_party/heap/mmtk/mmtk.cc:120]: OK. I would suggest adding some comments to explain this. Calling allocation semantics as 'space' and assigning it to magic numbers is confusing already, and there is another allocation space.

Comment (wenyuzhao) [reply v8/third_party/heap/mmtk/mmtk.cc:124]: Done

**Review by qinsoon — APPROVED** [review]

**Comment by qinsoon** [comment]
Please fix the build before merging. @wenyuzhao

**Comment by wenyuzhao** [comment]
Hi @qinsoon, could you please review my new changes to the CI scripts?

After this PR, v8-binding can run all the v8 tests with NoGC. So I updated the CI scripts as well, to run all of the v8 tests in CI.

**Comment by qinsoon** [comment]
Probably @javadamiri can review the PR. He is more familiar in this part than me. One question from me is that this PR was reviewed for space check changes, is there any reason to update V8 version and update test scripts in this PR?

**Review by javadamiri — APPROVED** [review]
LGTM

**Comment by wenyuzhao** [comment]
> Probably @javadamiri can review the PR. He is more familiar in this part than me. One question from me is that this PR was reviewed for space check changes, is there any reason to update V8 version and update test scripts in this PR?

I updated the tests just because after this PR, mmtk-v8 can pass all the v8's tests with NoGC. So we don't need to only run a subset of the tests, instead we can run all of them in CI.

**Pull request merged**